### ISSUE TITLE
Pod target for Intents Extension removes frameworks from 'Embed Pods Frameworks' phase

### ISSUE THREAD

**Issue by JanC** [CocoaPods/CocoaPods#8802]
* [x] I've read and understood the [*CONTRIBUTING* guidelines and have done my best effort to follow](https://github.com/CocoaPods/CocoaPods/blob/master/CONTRIBUTING.md).

# Report
Adding a `Podfile` entry for a `Intents Extension` target removes the frameworks needed by a `Notification Service Extension` target from the `[CP] Embed Pods Frameworks` phase.

## What did you do?
Create a Xcode project with the targets
- iOS App: `ExampleApp`
- Notification service extension: `NotificationServiceExtension`
- Intents service extension : `SiriIntentExtension`

Podfile:
```ruby
use_frameworks!
platform :ios, '11.0'

target 'ExampleApp' do
  pod 'Alamofire'
end

target 'NotificationServiceExtension' do
  pod 'DateToolsSwift'
end

# this is causing trouble
target 'SiriIntentExtension' do
  pod 'Alamofire'
end
```

ℹ Please replace these two lines with what you did.  
`pod install`

## What did you expect to happen?
The `ExampleApp`  `[CP] Embed Pods Frameworks` phase should contain the `.frameworks` of all target dependencies
<img width="760" alt="Screenshot_2019-05-14_at_11_18_34" src="https://user-images.githubusercontent.com/707703/57686635-39b73700-763a-11e9-840d-c75532d2fbdf.png">

## What happened instead?

The `ExampleApp` `[CP] Embed Pods Frameworks` phase does not include the  `DateToolsSwift.framework`  (needed by `NotificationServiceExtension`)  in the `[CP] Embed Pods Frameworks` phase. 

When building the app, the `Frameworks` folder of the `ExampleApp.app` do not include the needed framworks (in this case `DateToolsSwift`)

<img width="278" alt="Screenshot 2019-05-14 at 11 23 57" src="https://user-images.githubusercontent.com/707703/57687011-f1e4df80-763a-11e9-887e-be8e6395b0f0.png">

This causes the `NotificationServiceExtension` to fail during runtime as it cannot find its needed dynamic libraries.

<img width="763" alt="Screenshot_2019-05-14_at_11_18_58" src="https://user-images.githubusercontent.com/707703/57686646-3de35480-763a-11e9-929f-d124b8132c54.png">

## Solutions & Workarounds

I noticed that this happens if the `SiriIntentExtension` is **after** the `NotificationServiceExtension` entry in the Podfile. So If I move it above the notificaton extension target, it works correctly:

Podfile:
```ruby
use_frameworks!
platform :ios, '11.0'

target 'ExampleApp' do
  pod 'Alamofire'
end

# this needs to be before NotificationServiceExtension
target 'SiriIntentExtension' do
  pod 'Alamofire'
end

target 'NotificationServiceExtension' do
  pod 'DateToolsSwift'
end
```

Another workaround is to add explicitly the pod from the  `NotificationServiceExtension` to the Main app's target even if it's not needed there

```ruby
target 'ExampleApp' do
  pod 'Alamofire'
  pod 'DateToolsSwift'
end
```

### Stack

```
   CocoaPods : 1.6.1
        Ruby : ruby 2.3.7p456 (2018-03-28 revision 63024) [universal.x86_64-darwin18]
    RubyGems : 2.5.2.3
        Host : Mac OS X 10.14.4 (18E226)
       Xcode : 10.1 (10B61)
         Git : git version 2.17.2 (Apple Git-113)
Ruby lib dir : /System/Library/Frameworks/Ruby.framework/Versions/2.3/usr/lib
Repositories : goinstant - https://github.com/goinstant/pods-specs-public @ 8fc9da7f193f0c4d07b9d287b9237565cb2f7d3d
               master - https://github.com/CocoaPods/Specs.git @ 5a86534b32c6d75025683531ab25578fefa51b3f
               touch-tech-payments - https://github.com/Touch-Tech-Payments/Specs.git @ b9aafbaf5e47cf8767bbc36ddbeaf15e0a772992
```

### Installation Source

```
Executable Path: /usr/local/bin//pod
```

### Plugins

```
cocoapods-deintegrate : 1.0.4
cocoapods-plugins     : 1.0.0
cocoapods-search      : 1.0.0
cocoapods-stats       : 1.1.0
cocoapods-trunk       : 1.3.1
cocoapods-try         : 1.1.0
```

### Podfile

```ruby
use_frameworks!
platform :ios, '11.0'

target 'ExampleApp' do
  pod 'Alamofire'
end

target 'NotificationServiceExtension' do
  pod 'DateToolsSwift'
end

# this is causing trouble
target 'SiriIntentExtension' do
  pod 'Alamofire'
end
```

**Comment by dnkoutso** [comment]
Thanks for report @JanC ! We will need to investigate this.

**Comment by dnkoutso** [comment]
Actually @JanC can you try latest master version? I believe this is fixed by https://github.com/CocoaPods/CocoaPods/pull/8758

**Comment by JanC** [comment]
I did the following to install from master
```
gem uninstall cocoapods
gem specific_install -l https://github.com/CocoaPods/CocoaPods.git -b master
...
pod --version
1.7.0.rc.1
```

Then in my sample project

```
pod deintegrate
pod install
```

It seems that now non of the frameworks is in the `[CP] Embed Pods Frameworks` phase

<img width="1033" alt="Screenshot 2019-05-14 at 20 40 17" src="https://user-images.githubusercontent.com/707703/57723275-92adbc00-7688-11e9-97f7-1440957b541d.png">

I tried both with 
```ruby
use_frameworks!
platform :ios, '11.0'

target 'ExampleApp' do
  pod 'Alamofire'
end
```

and 
```ruby
use_frameworks!
platform :ios, '11.0'

target 'ExampleApp' do
  pod 'Alamofire'
end

 target 'NotificationServiceExtension' do
   pod 'DateToolsSwift'
 end

 # this is causing trouble
 target 'SiriIntentExtension' do
   pod 'Alamofire'
 end
```

I can confirm though that it's a regression as `cocoapods -v 1.5.3`  works as expected

**Comment by dnkoutso** [comment]
Can you please upload the sample?

**Comment by JanC** [comment]
sure

[ExampleApp.zip](https://github.com/CocoaPods/CocoaPods/files/3179345/ExampleApp.zip)

**Comment by dnkoutso** [comment]
It worked for me. can you please not look at the input/output paths since those were replaced with xcfilelists?

Look into `Pods-ExampleApp-frameworks.sh` file and see:

```
if [[ "$CONFIGURATION" == "Debug" ]]; then
  install_framework "${BUILT_PRODUCTS_DIR}/Alamofire/Alamofire.framework"
  install_framework "${BUILT_PRODUCTS_DIR}/DateToolsSwift/DateToolsSwift.framework"
fi
if [[ "$CONFIGURATION" == "Release" ]]; then
  install_framework "${BUILT_PRODUCTS_DIR}/Alamofire/Alamofire.framework"
  install_framework "${BUILT_PRODUCTS_DIR}/DateToolsSwift/DateToolsSwift.framework"
fi
```

I confirmed the list looks buggy in 1.7.0.rc.1 so the bug was fixed by the PR I mentioned.

**Comment by dnkoutso** [comment]
I will re-open this issue btw if we find out something else is wrong!

**Comment by dnkoutso** [comment]
This will be fixed with 1.6.2 btw and 1.7.0.rc.2!

**Comment by JanC** [comment]
ok confirming that it works with the master version 👍 All Frameworks are copied as expected

> those were replaced with xcfilelists
oh sorry, I didn't know that

> This will be fixed with 1.6.2 btw and 1.7.0.rc.2!
great! Thanks for looking into this
